app/services/migration/migrators/application_state.rb

```diff
@@ -0,0 +1,37 @@
+module Migration::Migrators
+  class ApplicationState < Base
+    class << self
+      def record_count
+        ecf_participant_profile_states.count
+      end
+
+      def model
+        :application_state
+      end
+
+      def dependencies
+        %i[application]
+      end
+
+      def ecf_participant_profile_states
+        Migration::Ecf::ParticipantProfileState
+          .joins(participant_profile: :teacher_profile)
+          .includes(:cpd_lead_provider)
+          .where(teacher_profile: { user_id: User.ecf_users.pluck(:id) })
```

Comment:
should the ecf users be a scope on the ECF User model, feels a bit odd to read into another migrator to get that query 🤔 happy to chat through

Reply:
I almost did this when I implemented it originally, but I reasoned that since the migrators depend on each other anyway it makes sense. Happy to change it to a scope on the `User` if you feel differently though

Reply:
since this is short lived code, its ok to keep as is for now